demo-linux/demo.cpp

```diff
@@ -130,12 +130,12 @@ void writeToEEPROM()
     string orig = bufToString(buf, PAGE_SIZE);
     printf("Original contents of EEPROM page %d: [%s]\n", USER_DATA, orig.c_str());
 
-    // generate new page contents
+    // generate new page contents 
     string writeString = "Wasatch.VCPP demo at " + timestamp();
 
     // write the EEPROM page
     printf("Overwriting page contents with: [%s]\n", writeString.c_str());
-    strncpy((char*)buf, writeString.c_str(), writeString.length());
+    strncpy((char*)buf, writeString.c_str(), PAGE_SIZE);
```

Comment:
Yes this prevents buffer overrun, but it is still vulnerable to non-termination. Use `PAGE_SIZE-1` to guarantee buff has a null terminator.